src/memory/memory.ts

```diff
@@ -30,37 +30,36 @@ class Memory {
     this.data.set(address, value);
   }
 
-  get(address: Relocatable) {
+  get(address: Relocatable): MaybeRelocatable {
     const value = this.data.get(address);
     if (value === undefined) {
       throw new UnknownAddressError();
     }
+    return value;
   }
 }
 
-class MemorySegmentManager {
-  segmentSizes: Record<number, number>;
+export class MemorySegmentManager {
+  segmentSizes: Record<Uint, Uint>;
   memory: Memory;
 
-  constructor(segmentSizes: Record<number, number>, memory: Memory) {
-    this.segmentSizes = segmentSizes;
-    this.memory = memory;
+  constructor() {
+    this.segmentSizes = {};
+    this.memory = new Memory();
```

Comment:
I think we'd want to be able to pass an existing memory, so as to wrap it?
Especially if we use it at the end of a program